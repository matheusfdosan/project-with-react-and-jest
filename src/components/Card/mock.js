export const mockCard = {
  id: 1,
  image: "https://fakestoreapi.com/img/81fPKd-2AYL._AC_SL1500_.jpg",
  title: "Fjallraven - Foldsack No. 1 Backpack, Fits 15 Laptops",
  description:
    "Your perfect pack for everyday use and walks in the forest. Stash your laptop (up to 15 inches) in the padded sleeve, your everyday",
  price: "$109.95",
  rating: {
    rate: "3.9",
  },
}
