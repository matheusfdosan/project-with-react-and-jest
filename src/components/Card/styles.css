.card {
  border-radius: 0.4rem;
  overflow: hidden;
  border: 0.1rem solid #37373788;
  padding: 1rem 2rem 2rem 2rem;
  cursor: pointer;
}

.card:hover {
  background-color: #37373728;
}

.card > .illustration {
  width: 100%;
  height: 35rem;
  padding: 3rem;
  object-fit: contain;
  mix-blend-mode: multiply;
}

.card-description > h3 {
  font-family: "Playir Display", serif;
  font-size: 2rem;
  margin-bottom: 1rem;
}

.card-description > p {
  font-weight: bold;
  font-size: 1.4rem;
  margin-bottom: 2rem;
  color: #37373788;
  height: 4rem;
  display: -webkit-box;
  -webkit-line-clamp: 2;
  -webkit-box-orient: vertical;
  text-overflow: ellipsis;
  overflow: hidden;
}

.card-footer {
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.card-footer > span:last-child {
  display: flex;
  align-items: center;
  gap: 0.5rem;
}

.card-footer > span:last-child > img {
  width: 2rem;
}
