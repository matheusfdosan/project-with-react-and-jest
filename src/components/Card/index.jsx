import React from "react"
import "./styles.css"

export default function Card({ data, onClickCard }) {
  const { id, image, title, description, price, rating } = data

  const handleClickCard = () => {
    onClickCard(data)
  }

  return (
    <div className="card" id={`card-number-${id}`} onClick={handleClickCard}>
      <img src={image} alt="illustration" className="illustration" />
      <div className="card-description">
        <h3>{title}</h3>
        <p>{description}</p>

        <div className="card-footer">
          <span>${price}</span>
          <span>
            <img
              src="https://www.iconpacks.net/icons/2/free-star-icon-2768-thumb.png"
              alt="star"
            />{" "}
            {rating.rate}
          </span>
        </div>
      </div>
    </div>
  )
}
