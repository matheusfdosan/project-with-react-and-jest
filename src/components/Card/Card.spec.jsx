import React from "react"
import "@testing-library/jest-dom"
import { fireEvent, render, screen } from "@testing-library/react"
import Card from "."
import { mockCard } from "./mock.js"

describe("<Card />", () => {
  it("should render component correctly", () => {
    render(<Card data={mockCard} />)

    expect(screen.getByAltText("illustration")).toHaveAttribute(
      "src",
      mockCard.image
    )

    expect(
      screen.getByRole("heading", { name: mockCard.title })
    ).toBeInTheDocument()

    expect(screen.getByText(mockCard.description)).toBeInTheDocument()

    expect(screen.getByText(`$${mockCard.price}`)).toBeInTheDocument()

    expect(screen.getByText(mockCard.rating.rate)).toBeInTheDocument()
  })

  it("should show modal on click", () => {
    const fn = jest.fn()
    const {container} = render(<Card onClickCard={fn} data={mockCard} />) // uma maneira de capiturar todo o componente. container.firstChild

    fireEvent.click(container.firstChild)
    expect(fn).toHaveBeenCalledTimes(1)
  })

  it("should match snapshot", () => {
    const fn = jest.fn()
    const { container } = render(<Card data={mockCard} onClickCard={fn}/>) 

    expect(container.firstChild).toMatchSnapshot() 
    // Testa se o componente está igual à Snapshot
  })
})
