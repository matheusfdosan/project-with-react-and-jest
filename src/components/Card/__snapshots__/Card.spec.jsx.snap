// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should match snapshot 1`] = `
<div
  class="card"
  id="card-number-1"
>
  <img
    alt="illustration"
    class="illustration"
    src="https://fakestoreapi.com/img/81fPKd-2AYL._AC_SL1500_.jpg"
  />
  <div
    class="card-description"
  >
    <h3>
      Fjallraven - Foldsack No. 1 Backpack, Fits 15 Laptops
    </h3>
    <p>
      Your perfect pack for everyday use and walks in the forest. Stash your laptop (up to 15 inches) in the padded sleeve, your everyday
    </p>
    <div
      class="card-footer"
    >
      <span>
        $
        $109.95
      </span>
      <span>
        <img
          alt="star"
          src="https://www.iconpacks.net/icons/2/free-star-icon-2768-thumb.png"
        />
         
        3.9
      </span>
    </div>
  </div>
</div>
`;
